packages/agents-core/src/types/protocol.ts

```diff
@@ -167,20 +176,99 @@ export const ToolOutputText = SharedBase.extend({
   text: z.string(),
 });
 
+export type ToolOutputText = z.infer<typeof ToolOutputText>;
+
+const ImageDataObjectSchema = z.object({
+  data: z
+    .union([z.string(), z.instanceof(Uint8Array)])
+    .describe(
+      'Base64 image data, or raw bytes that will be base64 encoded automatically.',
+    ),
+  mediaType: z.string().optional(),
+});
+
+const ImageUrlObjectSchema = z.object({
+  url: z
+    .string()
+    .describe('Publicly accessible URL pointing to the image content'),
+});
+
+const ImageFileIdObjectSchema = z.object({
+  fileId: z
+    .string()
+    .describe('OpenAI file ID referencing uploaded image content'),
+});
+
+const ImageObjectSchema = z
+  .union([ImageDataObjectSchema, ImageUrlObjectSchema, ImageFileIdObjectSchema])
+  .describe('Inline image data or references to uploaded content.');
+
+const FileDataObjectSchema = z.object({
+  data: z
+    .union([z.string(), z.instanceof(Uint8Array)])
+    .describe(
+      'Base64 encoded file data, or raw bytes that will be encoded automatically.',
+    ),
+  mediaType: z
+    .string()
+    .describe('IANA media type describing the file contents'),
+  filename: z.string().describe('Filename associated with the inline data'),
+});
+
+const FileUrlObjectSchema = z.object({
+  url: z.string().describe('Publicly accessible URL for the file content'),
+  filename: z.string().optional(),
+});
+
+const FileIdObjectSchema = z.object({
+  id: z.string().describe('OpenAI file ID referencing uploaded content'),
+  filename: z.string().optional(),
+});
+
+const FileReferenceSchema = z
+  .union([
+    z.string().describe('Existing data URL or base64 string'),
+    FileDataObjectSchema,
+    FileUrlObjectSchema,
+    FileIdObjectSchema,
+  ])
+  .describe(
+    'Inline data (with metadata) or references pointing to file contents.',
+  );
+
+const zStringWithHints = <T extends string>(..._hints: T[]) =>
+  z.string() as unknown as z.ZodType<T | (string & {})>;
+
 export const ToolOutputImage = SharedBase.extend({
```

Comment:
This has been available since the initial release, but actually there had been no way to utilize this type. That said, we're going to change the data structure to fit the reality, so I'd go with a minor version bump.